medyator/wrappers/request_handler_wrapper.py

```
@@ -1,41 +1,108 @@
+TQuery = TypeVar("TQuery", bound=Query[Any]) # Ensure Query is generic
+TAsyncQuery = TypeVar("TAsyncQuery", bound=AsyncQuery[Any]) # Ensure AsyncQuery is generic
 
-TQuery = TypeVar("TQuery", bound=Query)
-TCommand = TypeVar("TCommand", bound=Command)
 TResponse = TypeVar("TResponse")
 
-
-class RequestHandlerBase(Protocol):
-    def __call__(self, request: Any, service_provider: ServiceProvider) -> Any:
+# Protocol for the __call__ method, now async
+class AsyncRequestHandlerCallable(Generic[TRequest, TResponse]): # type: ignore
```

Ruff (0.11.9)

Undefined name `TRequest`

(F821)